www/api/class/webService.class.php

```diff
@@ -347,36 +347,48 @@ public static function router(\Pimple\Container $dependencyInjector, $user, $isI
             static::sendResult("Method not found", 404);
         }
 
+        /* Execute the action */
+        try {
+            static::checkWebserviceAuthorization($wsObj, $action, $user, $isInternal);
+
+            static::updateTokenTtl();
+            $data = $wsObj->$action();
+            $wsObj::sendResult($data, 200, $resultFormat);
+        } catch (RestException $e) {
+            $wsObj::sendResult($e->getMessage(), $e->getCode());
+        } catch (Exception $e) {
+            $wsObj::sendResult($e->getMessage(), 500);
+        }
+    }
+
+    /**
+     * Check webservice authorization
```

Comment:
Be careful, your method doesn't only check authorization, it also does redirections.
I suggest to change the name of your class like this: isWebserviceHasAllAuthorizations and to return TRUE or FALSE.
Then, depending on the status returned by this method you do the redirection.